lightning/src/ln/chan_utils.rs

```diff
@@ -1607,6 +1613,62 @@ impl CommitmentTransaction {
 		&self.htlcs
 	}
 
+	/// Returns the index of the revokeable output in the built transaction, if any exists.
+	/// There are two cases where this may return `None`:
+	/// - The balance of the revokeable output is below the dust limit (only found on commitments
```

Comment:
the HTLC outputs on a commitment transaction as also revokable, could be called the `to_local` output as it's only balance output you can revoke when you own the commitment. beyond, I wonder if the first case isn't more when the balance of the revokable output is between dust limit and channel reserve, if it's below the dust limit it should be removed according to bolt3 commitment transaction construction and our `build_commitment_transaction`, I think

Reply:
> the HTLC outputs on a commitment transaction as also revokable, could be called the to_local output as it's only balance output you can revoke when you own the commitment.

Good point, added some clarification that it's the revokeable output sending funds to the broadcaster

> beyond, I wonder if the first case isn't more when the balance of the revokable output is between dust limit and channel reserve, if it's below the dust limit it should be removed according to bolt3 commitment transaction construction and our build_commitment_transaction, I think

Our channel reserve is required to be above the dust limit so I think this is fine?

Reply:
> Our channel reserve is required to be above the dust limit so I think this is fine?

It’s correct our channel is required to be above the dust limit (we check against `MIN_CHAN_DUST_LIMIT_SATOSHIS` in `OpenChannel` reception) though I was pointing you might have a `to_local_output` amount between dust_limit_satoshis and channel_reserve during early channel lifetime (e.g you received one HTLC above `dust_limit_satoshis` though under the 1% of the channel capacity, our default value iirc), though once channel_reserve is met it’s cannot go under. Comment was a bit confusing to me, nevermind to change it.